File: PNC-UniStay-FrontEnd-main/src/services/bookmarkService.js
// src/services/bookmarkService.js
import apiClient from './apiClient';

export async function addBookmark(postId) {
    await apiClient.post(`/bookmarks/${postId}`);
}

export async function removeBookmark(postId) {
    await apiClient.delete(`/bookmarks/${postId}`);
}

export async function getMyBookmarks() {
    const { data } = await apiClient.get('/bookmarks');
    // data.data trae la lista de PostResponse por tu GeneralResponse
    return data.data || [];
}

File: RailWay-Backend-main/src/main/java/com/dog/service/BookmarkService.java
package com.dog.service;

import com.dog.dto.response.PostResponse;

import java.util.List;
import java.util.UUID;

public interface BookmarkService {

    void addBookmark(UUID postId, String userEmail);

    void removeBookmark(UUID postId, String userEmail);

    List<PostResponse> getMyBookmarks(String userEmail);
}
